### PR TITLE
ci: Add support for Rails 8

### LINKED ISSUE #798: [REQUEST] support rails 8.0
rswag-api, rswag-specs, rswag-ui

Could not find compatible versions

Because rswag-api >= 2.14.0 depends on railties >= 5.2, < 8.0
  and rails >= 8.0.0 depends on railties = 8.0.0,
  rswag-api >= 2.14.0 is incompatible with rails >= 8.0.0.
So, because Gemfile depends on rails ~> 8.0.0
  and Gemfile depends on rswag-api ~> 2.15.0,
  version solving has failed.

### PR DESCRIPTION
**Pull request by hasghari**
## Problem
Rails 8 was released today and the latest gemspec is not compatible.

## Solution
Update the gemspec to allow for Rails 8 and add it to the test matrix in CI.

### Checklist
- [ ] Added tests
- [x] Changelog updated
- [ ] Added documentation to README.md
- [ ] Added example of using the enhancement into [test-app](https://github.com/rswag/rswag/tree/master/test-app)

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -19,14 +19,15 @@ jobs:
 
         # Rails 5.2 became EOL on 2022-06-01; kept for information only
         # Rails 6.0 became EOL on 2023-06-01; kept for information only
```
Comment (zhuravel) [inline .github/workflows/ruby.yml:21]:
```suggestion
        # Rails 6.0 became EOL on 2023-06-01; kept for information only
        # Rails 6.1 became EOL on 2024-10-01; kept for information only
```

**Comment by bilby91** [comment]
Looking forward to see this PR merged 🚀

**Comment by bilby91** [comment]
@romanblanco Any ETA for a new version ? Thanks!

**Comment by eiskrenkov** [comment]
+1, Rails 8 support was merged yesterday. Please, release the new version to rubygems!
CC @romanblanco

**Pull request merged**